src/miner_gateway_port.erl

```diff
@@ -79,6 +82,25 @@ handle_info({'DOWN', Ref, port, _Pid, Reason}, #state{port = Port, monitor = Ref
                    State#state.tcp_port),
     ok = miner_gateway_ecc_worker:reconnect(),
     {noreply, NewState};
+handle_info(gw_healthcheck, State) ->
+    Me = self(),
+    Healthcheck = fun() -> Output = os:cmd(gateway_bin() ++ " -c " ++ gateway_config_dir() ++ " info"), Me ! Output end,
```

Comment:
I'd use the following alternative to `os:cmd/1`, so that we can check the exit code: https://gist.github.com/xandkar/20ef60a5ec2faaa069d71ee0e073f88e